### PR TITLE
chapter_12

### PR DESCRIPTION
**Pull request by azimfaiz**

### CODE REVIEW

**Comment by azimfaiz** [comment]
ready

File: app/assets/stylesheets/password_resets.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the PasswordResets controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/password_resets.scss]: remove file that never ued

File: app/controllers/password_resets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    return if @user
+
+    flash[:danger] = "User not found"
+    redirect_to root_url
+  end
+
+  def valid_user
+    return if @user && @user.activated? && @user.authenticated? :reset, params[:id]
+    redirect_to root_url
```
Comment (saobv-2223) [inline app/controllers/password_resets_controller.rb]:
```suggestion
    return if @user && @user.activated? && @user.authenticated? :reset, params[:id]
    
    redirect_to root_url
```

File: app/controllers/password_resets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  end
+
+  def check_expiration
+    return unless @user.password_reset_expired?
+    flash[:danger] = "Password reset has expired."
+    redirect_to new_password_reset_url
```
Comment (saobv-2223) [inline app/controllers/password_resets_controller.rb]:
```suggestion
    return unless @user.password_reset_expired?
    
    flash[:danger] = "Password reset has expired."
    redirect_to new_password_reset_url
```

File: app/helpers/password_resets_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module PasswordResetsHelper
+end
```
Comment (saobv-2223) [inline app/helpers/password_resets_helper.rb]: remove file that never used

File: config/routes.rb [hunk 5]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'password_resets/new'
+  get 'password_resets/edit'
```
Comment (saobv-2223) [inline config/routes.rb]: uisng "

**Comment by azimfaiz** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**